dashboard/app/models/pd/teacher_application.rb

```diff
@@ -93,7 +93,7 @@ def validate_required_application_fields
     end
   end
 
-  validate :primary_email_must_match_user_email
+  validate :primary_email_must_match_user_email, unless: -> {primary_email.blank?}
```

Comment:
primary_email is already [required](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/pd/teacher_application.rb#L77), so there's no need to run this validation if it's missing. We'll already have an error